src/argilla/server/contexts/datasets.py

```diff
@@ -388,11 +401,31 @@ async def get_records_by_ids(
     return ordered_records
 
 
+async def _configure_query_relationships(
+    query: "Select", include_params: Optional["RecordIncludeParam"] = None
+) -> "Select":
+    if not include_params:
+        return query
+
+    if include_params.with_suggestions:
+        query = query.options(joinedload(Record.suggestions))
+
+    if include_params.with_all_vectors:
+        query = query.options(joinedload(Record.vectors))
+
+    elif include_params.with_some_vector:
```

Comment:
It is a product requirement to return some specific vector only? I believe it's adding some complexity that we can avoid only returning all vectors or no vectors at all.

/cc @frascuchon